src/test/java/teammates/test/driver/GmailServiceMaker.java

```diff
@@ -0,0 +1,118 @@
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.store.FileDataStoreFactory;
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.GmailScopes;
+
+/**
+ * Class that builds a Gmail service for use in Gmail API.
+ */
+final class GmailServiceMaker {
+
+    /** Global instance of the JSON factory. */
+    private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+    /** Global instance of the HTTP transport. */
+    private static final HttpTransport HTTP_TRANSPORT;
+
+    static {
+        try {
+            HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+        } catch (GeneralSecurityException | IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private final String username;
+    private final boolean shouldUseFreshCredentials;
+
+    GmailServiceMaker(String username) {
+        this(username, false);
+    }
+
+    GmailServiceMaker(String username, boolean shouldUseFreshCredentials) {
+        this.username = username;
+        this.shouldUseFreshCredentials = shouldUseFreshCredentials;
+    }
+
+    /**
+     * Build and return an authorized Gmail client service.
+     * @return an authorized Gmail client service
+     */
+    Gmail makeGmailService() throws IOException {
+        Credential credential = authorizeAndCreateCredentials();
+        return new Gmail.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential)
+                .setApplicationName("teammates")
+                .build();
+    }
+
+    /**
+     * Authorizes the user and creates an authorized Credential.
+     * @return an authorized Credential
+     */
+    private Credential authorizeAndCreateCredentials() throws IOException {
+        GoogleClientSecrets clientSecrets = loadClientSecretFromJson();
+
+        GoogleAuthorizationCodeFlow flow = buildFlow(clientSecrets);
+
+        if (shouldUseFreshCredentials) {
+            flow.getCredentialDataStore().delete(username);
+        }
+
+        if (flow.getCredentialDataStore().get(username) == null) {
+            System.out.println("Please login as: " + username);
+        }
+
+        return getCredentialFromFlow(flow);
+    }
+
+    private GoogleClientSecrets loadClientSecretFromJson() throws IOException {
+        InputStream in;
+        try {
+            in = new FileInputStream(new File(TestProperties.TEST_GMAIL_API_FOLDER, "client_secret.json"));
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("You need to setup your client_secret.json." + System.lineSeparator()
```

Comment:
Yea it does, but just like `Const.EOL` it also saves the property. Why do we need to have our constant if there is one built in was what I meant.